### ISSUE TITLE
redirect_uri is adding extra / in the end

### ISSUE THREAD

**Issue by kesavkolla** [smart-on-fhir/client-js#85]
I'm using the following code:

```JavaScript
FHIR.oauth2.authorize({
    "client_id": "4a76-96f3-b3ac8db7d02a",
    "scope": "launch",
    "redirect_uri": "https://myapp.myorg.com"
});
```

The FHIR Oauth provider in my case was very strict in matching the **redirect_uri**. In my current case there is no trailing slash registered with the server that's why I passed redirect_uri parameter without trailing slash.  The code is automatically adding trailing slash in redirect_uri which is causing issues with authorization.

Ideally if redirect_uri is passed via options it shouldn't be altered. What I found in **smart.ts**

```JavaScript
if (!redirectUri) {
         redirectUri = env.relative(".");
    } else {
        redirectUri = env.relative(redirectUri);
     }
```
even when the redirectUri has some value it's calling env.relative function which is adding a trailing slash.  I think the else part should not be called.

**Comment by vlad-ignatov** [comment]
Thanks @kesavkolla,

Typically `redirectUri` should be a relative path. However, you seem to have a very specific (and rare) use case. For that reason, absolute URLs should now be used as is which should solve your issue. The documentation for this option is also updated. This is not officially released yet, but there is new `dev` version. Can you give it a try and see if it works for you? If you use it as a package try `npm i fhirclient@dev`. Otherwise link to (or download from) https://combinatronics.com/smart-on-fhir/client-js/master/dist/build/fhir-client.js

**Comment by kesavkolla** [comment]
I do see document from browserless (https://docs.browserless.io/docs/playwright.html) which says it works with playwright and they connect using browserWSEndpoint. I was thinking such a similar thing for page

**Comment by vlad-ignatov** [comment]
I'm sorry but I don't understand what you mean. Did you intend to post this comment in another repository?

**Comment by kesavkolla** [comment]
Oh I'm sorry that is true. Wrong post here

**Comment by Lakkanna** [comment]
@vlad-ignatov  **npm install fhirclient@dev** throwing an error while installing
![image](https://user-images.githubusercontent.com/32791713/92219427-2f1c5300-eeb8-11ea-8314-b919c940a173.png)

**Comment by vlad-ignatov** [comment]
Sorry! That was a copy-paste error. Should be fixed now.

**Comment by vlad-ignatov** [comment]
The fix is now released in `2.3.5`. Feel free to reopen if it does not solve the issue.
